### PR TITLE
DAND-7- Update Amplitude and Segment Analytics.

### PR DESCRIPTION
**Pull request by OllaAshour**
**What does this PR do?**

Update Amplitude and Segment Analytics SDK to latest version and also other dependencies.

- Analytics (3.7.0)
- Expecta (1.0.6)
- OCHamcrest (7.1.2)
- OCMockito (5.1.2)
- Specta (1.0.7)

**Associated JIRA Tickets:**
https://segmentcontractors.atlassian.net/browse/DAND-7

### CODE REVIEW

File: Segment-Amplitude.podspec [hunk 1]
```diff
@@ -21,6 +21,6 @@ Pod::Spec.new do |s|
 
   s.source_files = 'Pod/Classes/**/*'
 
-  s.dependency 'Analytics', '~> 3.6' # support for iOS 7 +
-  s.dependency 'Amplitude-iOS', '~> 4.0'
+  s.dependency 'Analytics'
+  s.dependency 'Amplitude-iOS'
 end
```
Comment (briemcnally) [inline Segment-Amplitude.podspec]: @OllaAshour can we actually add the specific upgraded target versions back here.  The` Example/Podfile.lock` is just keeping track of what's there, but if a customer does a pod install, that lock isn't going to be looked at and will be overwritten by the latest version which will be pulled by the `.podspec`  unless we indicate a target version for the dependency. The path of just taking the "latest" (not specifying version) is leaving us open to incompatibilities when a partner makes a backwards incompatible change to their SDK.

**Review by briemcnally — CHANGES_REQUESTED** [review]
Left one comment, otherwise looks amazing! Thanks for the great work!

File: Segment-Amplitude.podspec [hunk 2]
```diff
@@ -21,6 +21,6 @@ Pod::Spec.new do |s|
 
   s.source_files = 'Pod/Classes/**/*'
 
-  s.dependency 'Analytics'
-  s.dependency 'Amplitude-iOS'
+  s.dependency 'Analytics', '~> 3.6' # support for iOS 7 +
```
Comment (briemcnally) [inline Segment-Amplitude.podspec]: Can we actually change this back to the dependencies we are trying to bump to: ie: Analytics (3.7.0) and Amplitude-iOS (4.8.2)?

**Comment by migs647** [comment]
Since been updated

**Pull request closed without merging**